Detectors/TRD/calibration/src/DCSProcessor.cxx

```diff
@@ -59,6 +59,43 @@ int DCSProcessor::process(const gsl::span<const DPCOM> dps)
     }
   }
 
+  // LB: check if all ChamberStatus/CFGtag DPs were sent by using a vector and then
+  // checking for duplicates, if new size is equal to 540 => all DPs were sent
+  std::vector<DPID> allChamberStatusDPsid;
+  std::vector<DPID> allCFGtagDPsid;
```

Comment:
I think per iteration a DP cannot be send more than once. So the `std::unique` below should not be needed. To be sure, you could just use a `std::set` here which in any case only holds a unique set of objects.
But I would rather use the `mPids` map which is already there. Can you not check this if a sufficient number of DPs of a certain type have been seen?

Reply:
I also think the DP cannot be duplicated, so I removed this part of the code and changed for a simple counter given the entries of `dps`. Since assuming duplicates adds some complexity/confusion to the code, I think it is better to go without it and check the logs if there are duplicates, then we can do a quick fix.

We cannot use the `mPids` to track if ChamberStatus/CFGtag are complete, because the map is filled with the configuration information. We also cannot track if all objects following a pattern were changed to true (after processing) because we need this information before calling the processor. Maybe you meant `mapin`? I changed the code so the check of pattern is done in the same loop as `mapin`.

See commit https://github.com/AliceO2Group/AliceO2/pull/12079/commits/c1e9bcaa8af2752225fb1fd1165e3715813d0243.